File: bronze/9506/c++/main.cpp
#include <iostream>
#include <string>

int main()
{

    int N = 0, result;

    std::string puts;
    while (N != -1) {
        std::cin >> N;
        if (N == -1) {
            break;
        }
        result = 1;
        puts = std::to_string(N) + " = 1";
        
        for (int i = 2; i < N; i++) {
            if (N % i == 0) {
                puts += " + " + std::to_string(i);
                result += i;
            }
        }
        
        if (result == N) {
            std::cout << puts << '\n';
        } else {
            std::cout << N << " is NOT perfect." << '\n';
        }
        
    }

    return 0;
}

File: bronze/9506/python/main.py

if (__name__ == "__main__"):
    N = 0
    while (True):
        N = int(input())
        if (N == -1):
            break
        puts = f"{N} = {1}"; result = 1
        for i in range(2, N):
            if (N % i == 0):
                puts += f" + {i}"
                result += i
        if (N == result):
            print(puts)
        else :
            print(f"{N} is NOT perfect.")


